stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt

```diff
@@ -527,6 +549,27 @@ public class MessageInputView : ConstraintLayout {
         }
     }
 
+    /**
+     * Creates an instance of [SuggestionListControllerListener].
+     *
+     * Used to modify attachment button visibility to discourage mixing
+     * commands and attachments since they are not compatible.
+     */
+    private fun createSuggestionsListControllerListener(): SuggestionListControllerListener =
+        object : SuggestionListControllerListener {
+            override fun onSuggestionListUiVisibilityChanged(isVisible: Boolean) {
```

Comment:
Could we potentially wrap this in a class that's called `DefaultSuggestionListControllerListener` and then hold the values/properties `isSuggestionListUiVisible` and `inputContainsCommands` there?

It seems like we're adding two variables to the message input view just to keep them around to disable the UI, but I'd much rather receive an emission from the listener here that's a composite of the checks we have:

E.g. 
```
          override fun onSuggestionListStateChanged(shouldEnableAttachments: Boolean) {
                binding.attachmentsButton.isEnabled = shouldEnableAttachments
            }
```

Just have a single function (or a lambda param) that emits if the attachments should be enabled or disabled. This way we don't have to calculate anything ourselves and the logic is featured in one place.